src/content/en/updates/2018/03/present-web-pages-to-secondary-attached-displays.md

```diff
@@ -0,0 +1,268 @@
+'https://example.com']` as this is not relevant there.
+
+We can now monitor presentation display availability and for instance toggle a
+"Present" button visibility based on presentation displays availability. Note
+that we can also decide to always show this button. 
+
+Caution: the browser may use more energy while the `availability` object is alive
+and actively listening for presentation display availability changes. Please
+use it with caution in order to save energy on mobile.
+
+    presentationRequest.getAvailability()
+    .then(availability => {
+      console.log('Available presentation displays: ' + availability.value);
+      availability.addEventListener('change', function() {
+        console.log('> Available presentation displays: ' + availability.value);
+      });
+    })
+    .catch(error => {
+      console.log('Presentation availability not supported, ' + error.name + ': ' +
+          error.message);
+    });
+
+Showing a presentation display prompt requires a user gesture such as a click
+on a button. So let’s call `presentationRequest.start()` on a button click for
+instance and wait for the promise to resolve once user has selected a
+presentation display (.e.g. a secondary attached display in our use case).
+
+    function onPresentButtonClick() {
+      presentationRequest.start()
+      .then(connection => {
+        console.log('Connected to ' + connection.url + ', id: ' + connection.id);
+      })
+      .catch(error => {
+        console.log(error);
+      });
+    }
+
+List presented to the user may also include remote endpoints such as Chromecast
+devices if you’re connected to a network advertising some.
+
+<figure>
+  <img src="/web/updates/images/2018/03/present-web-pages-to-secondary-attached-displays/picker.png"
+    alt="Presentation Display Picker"/>
+</figure>
+
+When promise resolves, the web page at the `PresentationRequest` object URL is
+presented to the attached display user has previously picked. And voila!
+
+We can now go further and monitor "close" and "terminate" events quite easily
+as you can see below. Note that it is possible to reconnect to a "closed"
+`presentationConnection` with `presentationRequest.reconnect(presentationId)`
+where `presentationId` is the ID of the previous `presentationRequest` object.
+
+    function onCloseButtonClick() {
+      // Disconnect presentation connection but will allow reconnection.
+      presentationConnection.close();
+    }
+    
+    presentationConnection.addEventListener('close', function() {
+      console.log('Connection closed.');
+    });
+    
+    
+    function onTerminateButtonClick() {
+      // Stop presentation connection for good.
+      presentationConnection.terminate();
+    }
+    
+    presentationConnection.addEventListener('terminate', function() {
+      console.log('Connection terminated.');
+    });
+
+## Communicate with the page
+
+That’s nice but how do I pass messages between my controller page (the one
```

Comment:
'That's nice' > 'Now you're thinking, that's nice'

Reply:
Done.